File: java/kode program/Produk.java
public class Produk {
    protected int id; // ID produk
    protected String nama;// Nama produk
    protected double harga;// Harga produk
    protected String kategori; // Kategori produk

    // Constructor dengan parameter untuk inisialisasi atribut
    public Produk(int id, String nama, double harga, String kategori) {
        this.id = id; this.nama = nama; this.harga = harga; this.kategori = kategori;
    }

    // Method untuk menampilkan data produk dalam format tabel
    public void tampilkan() {
        System.out.printf("%5d%15s%12.0f%12s", id, nama, harga, kategori);
    }

    // Getter & Setter untuk mengakses dan mengubah atribut
    public int getId() { return id; }
    public void setId(int id) { this.id=id; }

    public String getNama() { return nama; }
    public void setNama(String nama) { this.nama=nama; }

    public double getHarga() { return harga; }
    public void setHarga(double harga) { this.harga=harga; }

    public String getKategori() { return kategori; }
    public void setKategori(String kategori) { this.kategori=kategori; }
}

File: python/kode program/produk.py
class Produk:
    def __init__(self, id, nama, harga, kategori):
        self.__id = id # ID produk (private)
        self.__nama = nama# Nama produk (private)
        self.__harga = harga# Harga produk (private)
        self.__kategori = kategori  # Kategori produk (private)

    # Getter & Setter untuk mengakses dan mengubah atribut private
    def get_id(self): return self.__id
    def set_id(self, value): self.__id = value

    def get_nama(self): return self.__nama
    def set_nama(self, value): self.__nama = value

    def get_harga(self): return self.__harga
    def set_harga(self, value): self.__harga = value

    def get_kategori(self): return self.__kategori
    def set_kategori(self, value): self.__kategori = value
